### ISSUE TITLE
[Bug]: Factor scores at listwise method

### ISSUE THREAD

**Issue by smhtpz** [jasp-stats/jasp-issues#3095]
### JASP Version

0.19.1

### JASP Module

Factor

### What analysis are you seeing the problem on?

Exploratory Factor Analysis - Factor Score

### What OS are you seeing the problem on?

Windows 11

### Bug Description

If you have missing values and you select liswise in exploratory factor analysis, when you try to add the Factor Score to the dataset, it groups the missing data at the end of the dataset. In this case, people are not assigned their own factor scores.
You can see 2 screenshot. One of them first rows of factor scores. Listwise method on the left and pairways method on the right are the 4 factor scores. In the second screenshot you can see the last rows of the data set. 

![image](https://github.com/user-attachments/assets/5697ac29-1205-47ae-a714-2010ff592608)

![image](https://github.com/user-attachments/assets/ba5cdf7f-493f-4ff2-8599-7a1a88ae1eae)

### Expected Behaviour

We expect each person to be assigned their own factor score as in Pairwise. I did not try with other analysis but maybe other analysis have errors like this.

### Steps to Reproduce

1. Add a factorable dataset with missings
2. Select Factor>Explarotary Factor Analysis
3. Select items build your model. It's already selected default "Pairwise Missing Method"
4. Select Add FA scores to data and change prefix, maybe "FAx"
5. After that copy analysis
6. Select "Listwise" and Add FA scores with different Prefix. End of the process you need to have 2 set of factor scores for listwise and pairwise.
7. You can see the bug at the end of the data or any missing cases.

### More Debug Information

-------- Application Info --------
JASP Version: JASP 0.19.1
Build Branch: lockfiletesting
Build Date: Sep  9 2024 19:00:04 (Netherlands)
Last Commit: 8ac10c9c7aec029fdd695262d7bb4c3742a8ec89

-------- Basic Info --------
Operating System: Windows 11 Version 24H2
Product Version: 11
Kernel Type: winnt
Kernel Version: 10.0.26100
Architecture: x86_64
Install Path: C:/Program Files (x86)/JASP
Platfotm Name: windows
System Local: tr_TR

-------- Extra Info --------
Current code page
Active code page: 857
Active code page: 65001

Host Name:                     ---
OS Name:                       Microsoft Windows 11 Education
OS Version:                    10.0.26100 N/A Build 26100
OS Manufacturer:               Microsoft Corporation
OS Configuration:              Standalone Workstation
OS Build Type:                 Multiprocessor Free
Registered Owner:              User
Registered Organization:       
Product ID:                    00328-10000-00001-AA378
Original Install Date:         25 Kas 2024, 10:02:13
System Boot Time:              16 Ara 2024, 15:29:37
System Manufacturer:           HP
System Model:                  HP EliteDesk 800 G6 Tower PC
System Type:                   x64-based PC
Processor(s):                  1 Processor(s) Installed.
                               [01]: Intel64 Family 6 Model 165 Stepping 5 GenuineIntel ~2904 Mhz
BIOS Version:                  HP S01 Ver. 02.20.01, 29 Ağu 2024
Windows Directory:             C:\WINDOWS
System Directory:              C:\WINDOWS\system32
Boot Device:                   \Device\HarddiskVolume1
System Locale:                 tr;Türkçe
Input Locale:                  tr;Türkçe
Time Zone:                     (UTC+03:00) İstanbul
Total Physical Memory:         16.160 MB
Available Physical Memory:     5.662 MB
Virtual Memory: Max Size:      20.512 MB
Virtual Memory: Available:     6.501 MB
Virtual Memory: In Use:        14.011 MB
Page File Location(s):         C:\pagefile.sys
Domain:                        WORKGROUP
Logon Server:                  \\--
Hotfix(s):                     3 Hotfix(s) Installed.
                               [01]: KB5048162
                               [02]: KB5048667
                               [03]: KB5049685
Network Card(s):               1 NIC(s) Installed.
                               [01]: Intel(R) Ethernet Connection (11) I219-LM
                                     Connection Name: Ethernet
                                     DHCP Enabled:    Yes
                                     DHCP Server:     ---
                                     IP address(es)
                                     [01]: ---
                                     [02]: ---
                                     [03]: ---
                                     [04]: ---
Virtualization-based security: Status: Running
                               Required Security Properties:
                               Available Security Properties:
                                     Base Virtualization Support
                                     DMA Protection
                                     UEFI Code Readonly
                                     SMM Security Mitigations 1.0
                                     Mode Based Execution Control
                                     APIC Virtualization
                               Services Configured:
                                     Hypervisor enforced Code Integrity
                               Services Running:
                                     Credential Guard
                                     Hypervisor enforced Code Integrity
                               App Control for Business policy: Enforced
                               App Control for Business user mode policy: Off
                               Security Features Enabled:
Hyper-V Requirements:          A hypervisor has been detected. Features required for Hyper-V will not be displayed.

### Final Checklist

- [X] I have included a screenshot showcasing the issue, if possible.
- [ ] I have included a JASP file (zipped) or data file that causes the crash/bug, if applicable.
- [X] I have accurately described the bug, and steps to reproduce it.

**Comment by tomtomme** [comment]
@smhtpz 
thx for the report. Any chance you could re-test this with version 0.19.2?

@juliuspfadt 
have you seen this before?

**Comment by smhtpz** [comment]
Yes I tried right now. 0.19.2 have same bug. Also PCA have same bug.

**Comment by juliuspfadt** [comment]
Good find @smhtpz. I haven't seen it before but it makes sense that it doesnt work well yet. . I will have to talk to the other developers on how to fix it.
